wasm/module_test.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package wasm_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-interpreter/wagon/wasm"
+)
+
+func TestReadModule(t *testing.T) {
+	fnames, err := filepath.Glob("testdata/*wasm")
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, fname := range fnames {
+		raw, err := ioutil.ReadFile(fname)
+		if err != nil {
+			t.Error(err)
+			continue
+		}
+		r := bytes.NewReader(raw)
+		_, err = wasm.ReadModule(r, nil)
+		if err != nil {
+			t.Errorf("error reading module %q: %v", fname, err)
+			continue
+		}
```

Comment:
If you wanted to, you could modify line above to `m, err = wasm.ReadModule(r, nil)`, and add a quick test here that `m != nil`:

```Go
if m == nil {
	t.Errorf("wasm.ReadModule of %q return nil module and nil error", fname)
}
```

(Then, the `continue` above would be needed.)